drivers: entropy: stm32: Keep RNG clock running during PKA operation

gfroissard opened the pull request:
When using STM32 PKA hardware accelerator the RNG's clock is needed. This PR makes entropy (RNG module) not turning off its clock if PKA is enabled. Before this PR, it was done only for `STM32WBAX` family.

mathieuchopstm commented on drivers/entropy/entropy_stm32.c:
```diff
@@ -140,12 +140,15 @@ static int entropy_stm32_suspend(void)
-#ifdef CONFIG_SOC_SERIES_STM32WBAX
-	uint32_t wait_cycles, rng_rate;
-
+#if defined(PKA)
```
This will hang on N6 if PKA peripheral is not clocked.

`if (__HAL_RCC_PKA_IS_CLK_ENABLED())` check before `LL_PKA_IsEnabled()`

mathieuchopstm reviewed (dismissed):
Needs some testing (notably N6) but LGTM otherwise.

erwango commented on drivers/entropy/entropy_stm32.c:
```diff
@@ -140,12 +140,15 @@ static int entropy_stm32_suspend(void)
-	uint32_t wait_cycles, rng_rate;
-
-	if (LL_PKA_IsEnabled(PKA)) {
+#if defined(PKA)
```
Please add a comment to explicit that we're talking about the use of PKA "outside of zephyr" as not supported yet.
(otherwise this wouldn't be implemented this way).

erwango reviewed (changes requested):
Thanks for sharing this fix, some tweaks required.

etienne-lms commented on drivers/entropy/entropy_stm32.c:
```diff
@@ -140,12 +140,16 @@ static int entropy_stm32_suspend(void)
+#if defined(PKA)
+	if (__HAL_RCC_PKA_IS_CLK_ENABLED() && LL_PKA_IsEnabled(PKA)) {
+		/* PKA needs RNG clock, so exit here if in use */
 		return 0;
```
Unlock semaphore for WBX and H7 dual core devices?
```suggestion
#if defined(CONFIG_SOC_SERIES_STM32WBX) || defined(CONFIG_STM32H7_DUAL_CORE)
		z_stm32_hsem_unlock(CFG_HW_RNG_SEMID);
#endif /* CONFIG_SOC_SERIES_STM32WBX || CONFIG_STM32H7_DUAL_CORE */
		return 0;
```
**(edited) actually only needed for WBXX.**

gfroissard committed:
drivers: entropy: stm32: Keep RNG clock running during PKA operation

If PKA is running, it needs the RNG clock to properly works.
Don't turn it off when RNG computation is over.

Signed-off-by: Gaétan Froissard <gaetan.froissard@marshmallow.kids>
